src/lib/components/DynamicTable/DynamicTable.tsx

```diff
@@ -0,0 +1,122 @@
+import type { AriaAttributes, PropsWithChildren, RefObject } from "react";
+import {
+  useState,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useCallback,
+} from "react";
+
+import type { RowData, Table } from "@tanstack/react-table";
+import classNames from "classnames";
+
+import { BREAKPOINTS } from "@/constants";
+import { Placeholder } from "@/lib/elements";
+import "./DynamicTable.scss";
+
+export type DynamicTableProps = PropsWithChildren<{ className?: string }>;
+
+export const DynamicTable = ({
```

Comment:
Let's add some basic documentation explaining what's this component is for and what makes it different from a regular table.